game.cpp

```diff
@@ -0,0 +1,265 @@
+
+#include <iostream>
+#include <vector>
+using namespace std;
+
+
+class Grid
+{
+public:
+  Grid() {
+    row=3;
+    col=3;
+    player=1;
+    line_length=3;
+    // 0 means empty cell, X=1, O=2
+    board=vector<vector<int> > (row, vector<int> (col, 0));
+  }
+
+  virtual ~Grid(){}
+
+  int get_player(){
+    return player;
+  }
+
+private:
+  vector<vector<int> > board;
+  int row, col;
+  //1 is X and goes first, 2 is O
+  int player;
+  int line_length;
+
+  void set_cell(int r, int c, int value)
+  {
+    //If (r, c) is a valid cell, sets it to value
+    //Otherwise, logs an error
+    if (0<=r and r<row and 0<=c and c<col){
+      board[r][c]=value;
+    }
+    else {
+      cerr<<"Attempt to set invalid cell ("<<r<<", "<<c<<") to "<<value<<endl;
+    }
+  }
+
+  int find_cell(int cell_num) {
+    // The player inputs a number from 1 to 9, corresponding to a cell:
+    /*
+    0|1|2
+    -+-+-
+    3|4|5
+    -+-+-
+    6|7|8
+    */
+    //This returns the value of the corresponding cell
+    return get(cell_num/row, cell_num%col);
+  }
+
+  char get_char(int cell_val) {
+    // Converts 0, 1 and 2 to the relevant character
+    switch (cell_val)
+    {
+    case 0:
+      return ' ';
+    case 1:
+      return 'X';
+    case 2:
+      return 'O';
+
+    default:
+      break;
+    }
+  }
+
+  void print() {
+    for (int i = 0; i < row; ++i)
+    {
+      for (int j = 0; j < col; ++j)
+      {
+        cout<<get_char(get(i, j));
+        if (j<col-1){
+          cout<<"|";
+        }
+      }
+      cout<<endl;
+      if (i<row-1)
+      {
+        cout<<"-----\n";
+      }
+    }
+  }
+
+  int get(int r, int c){
+    //Returns the cell (r, c)
+    //If these are invalid coordinates, logs an error and returns -1
+    if (0<=r and r<3 and 0<=c and c<3){
```

Comment:
Generally you've been using `row` and `column` as the sizes of your grid, but here you are using a hardcoded `3`.